python/pyproject.toml

```diff
@@ -73,7 +73,9 @@ zip-safe=false
 include-package-data=true
 
 [tool.setuptools.packages.find]
-where = ["."]
+include = ["pyarrow"]
+exclude = ["pyarrow/tests", "pyarrow."]
```

Comment:
this doesn't seem to have any effect, @jorisvandenbossche @pitrou any idea?
The other stray directories are filtered from the wheel now

Reply:
I believe the syntax is `pyarrow.tests`. What does `pyarrow.` refer to?

Reply:
there is a folder included in the wheel `pyarrow.` which is empty and I was trying to remove it. I'll try with `pyarrow.tests`

Reply:
it doesn't remove tests either

Reply:
We don't care about removing the tests, they are comparatively small and can serve to test PyArrow on a third-party machine.
```console
$ du -hs venv-3.10/lib/python3.10/site-packages/pyarrow
187M	venv-3.10/lib/python3.10/site-packages/pyarrow
$ du -hs venv-3.10/lib/python3.10/site-packages/pyarrow/tests/
4,0M	venv-3.10/lib/python3.10/site-packages/pyarrow/tests/
```

Reply:
what about cpp and header files, should we do something like pandas here:
https://github.com/pandas-dev/pandas/blob/main/pyproject.toml#L131

Reply:
Currently the files under `pyarrow/src/arrow/python`  are included on the wheel

Reply:
I'll stress that the problem we're trying to solve is that installing a PyArrow wheel creates top-level directories _outside_ of the `pyarrow` source tree. This is the urgency. Cleaning up the contents of `pyarrow` is a separate task, much less critical.

Reply:
I guess those are negligible and we still require to distribute them on the source distitrbution:
```
pyarrow/src $ du -hs .
728K	.
```

Reply:
> I'll stress that the problem we're trying to solve is that installing a PyArrow wheel creates top-level directories _outside_ of the `pyarrow` source tree.

ok, then, this is solved by this PR, are you ok with the current changes @pitrou ?